staging/src/k8s.io/apimachinery/pkg/util/rand/rand.go

```diff
@@ -18,63 +18,44 @@ limitations under the License.
 package rand
 
 import (
-	"math/rand"
-	"sync"
+	"math/rand/v2"
 	"time"
 )
 
-var rng = struct {
-	sync.Mutex
-	rand *rand.Rand
-}{
-	rand: rand.New(rand.NewSource(time.Now().UnixNano())),
-}
+var rng = rand.New(rand.NewPCG(uint64(time.Now().UnixNano()), 0))
 
 // Int returns a non-negative pseudo-random int.
 func Int() int {
-	rng.Lock()
-	defer rng.Unlock()
-	return rng.rand.Int()
+	return rng.Int()
 }
 
 // Intn generates an integer in range [0,max).
 // By design this should panic if input is invalid, <= 0.
 func Intn(max int) int {
-	rng.Lock()
-	defer rng.Unlock()
-	return rng.rand.Intn(max)
+	return rng.IntN(max)
 }
 
 // IntnRange generates an integer in range [min,max).
 // By design this should panic if input is invalid, <= 0.
 func IntnRange(min, max int) int {
-	rng.Lock()
-	defer rng.Unlock()
-	return rng.rand.Intn(max-min) + min
+	return rng.IntN(max-min) + min
 }
 
 // IntnRange generates an int64 integer in range [min,max).
 // By design this should panic if input is invalid, <= 0.
 func Int63nRange(min, max int64) int64 {
-	rng.Lock()
-	defer rng.Unlock()
-	return rng.rand.Int63n(max-min) + min
+	return rng.Int64N(max-min) + min
 }
 
 // Seed seeds the rng with the provided seed.
 func Seed(seed int64) {
-	rng.Lock()
-	defer rng.Unlock()
-
-	rng.rand = rand.New(rand.NewSource(seed))
+	rng = rand.New(rand.NewPCG(uint64(seed), 0))
```

Comment:
previously, this was threadsafe, but now concurrent calls to set a package variable could cause a data race with other calls to Seed or any of the functions that read / call methods on `rng`

you could resolve this by making rng an atomic.Pointer that readers can Load() and this function can Store(), or by making a package RWMutex readers can RLock/RUnlock and this function can Lock/Unlock

Reply:
Fixed by making `rng` an `atomic.Pointer`. Thank you for the review!